lib/models/cosmos_wallet.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:sacco/wallet.dart';
+
+class CosmosWallet extends BaseWalletDetails {
```

Comment:
we should avoid inheritance wherever possible and stick to composition. We could simply get rid of the `CosmosWallet` and `EthWallet` classes as they don't bring any additional value here

Reply:
We are using this inheritance to cater generic wallets in our UI. We are also reusing code from `BaseWalletDetails` in order to keep things under abstraction for the UI.

Reply:
you can leverage composition in following way:

```dart
 class CosmosWallet {
  final WalletDetails walletDetails;
  final Wallet wallet;
  final WalletType walletType;

  const CosmosWallet({
    required this.walletDetails,
    required this.wallet,
  }) : walletType = WalletType.cosmos;
}

...

class WalletDetails {
  final String walletAddress;
  final String walletAlias;

  const WalletDetails({
    required this.walletAddress,
    required this.walletAlias,
  });


}
```

I just have some problems with current naming, there's a lot of wallets right now, but we can work that out in a bit, when the final architecture outline is ready

Reply:
Two confusions:
- In that case, how will the UI render different kinds of wallets altogether? The object will be of different types then. Currently the UI list only knows that there are wallets and not types of wallets.
- Since `CosmosWallet` is a `Wallet` and not a part of a wallet so composition in this case seems a little off to me. What do you think?

Reply:
in order to support multiple wallets then we can hide it behind abstraction like this:
```
abstract class EmerisWallet {
  WalletDetails get walletDetails;
  WalletType get walletType;
}

class CosmosWallet implements EmerisWallet {
    ...
}
```

Regarding the `Wallet` object from Sacco library, this should not be part of the domain entity and should not be accessed directly from the UI, since it is a specific implementation, that could also be hidden behind abstraction to allow for easy swapping in the future, but this can be worked on later on, in a separate pr.

Reply:
This sounds cool. Yes the Wallet object from libraries need to be abstracted, they should not be accessed by the UI, which is why we will fix that in the architecture. I am, however, still not convinced about using composition instead of inheritance. In real world scenarios, it's about being a part of. In our case it's otherwise.

Reply:
With composition we allow for better code reusability and readability. if we stick to inheritance, we can end up in inheritance hell with multiple levels of `extends`. this makes hard to determine what hte code is actually doing without having to navigate trough multiple levels of classes. Here the `EmerisWallet` consists of `WalletDetails` and `WalletType` which defines that each wallet in our app should support having details of the wallet `i.e: public address and alias` as well as `walletType`

Reply:
Perfect. Now it makes sense. The `WalletDetails` is now not a part of `CosmosWallet`, it's inside the abstraction now. Am I getting it right now?

Reply:
yes, pretty much, though this abstraction is merely an interface, thus CosmosWallet , when implementing the abstraction will need to expose a walletDetails field

Reply:
Updated ✅